specs/speculative_specification.md

```diff
@@ -0,0 +1,67 @@
+By heavy use of tab-completion, users can explore what tools are available from their project without having to immediately resort to reading documentation. 
+
+The core of pyiron, and the most common object created from a project, is a `Job`.
+Jobs are the objects which actually perform a calculation by invoking their `run` command.
+Job objects bring together the core elements of pyiron: they provide an easy to use and easy to extend python interface; they support reproducibility by serializing the details of input and executed code(TODO) with output results; they facilitate analysis and collaboration by populating a database with their most important details; they are scalable by simply specifying which remote resource and how many cores to use.
+Importantly, they do all these things without heavy intervention from the user.
+Users simply create a job from a project, specify appropriate `input`, (optionally) specify `server` details for HPC calculations, and call the `run()` method.
+Under the hood, pyiron then handles all the serialization, registration with the database, and HPC queue and job status management.
+
+The most common type of job is one that wraps some third-party software to performa a calculation.
+In this case pyiron provides a simple framework for research management, ensuring that input and results are coupled, and making results easier to share, parse, and analyze.
+Pyiron supports Jupyter notebook environments, so users can easily assemble workflows (including visual analysis) in an easy-to-use pure-python environment. 
+However, pyiron also provides more sophisticated job types and has templates for various workflow abstractions.
+Thus, power-users can go beyond notebook-as-workflow to develop robust and shareable tools for their communities.
+
+Finally, pyiron is also highly customizable.
+Configuration for various queueing systems, as well as access to and resources for various third party codes are all possible in the pyiron resources directory.
+Other customization, e.g. whether to run with or without a database, is handled by the global `State` object.
+Pyiron also supports a variety of working environments, including local (e.g. your laptop), HPC clusters, and even accessing a variety of remote systems, doing calculations there, and collecting results back to another central system.
+
+
+# How pyiron -- guidance for developers
+
+Database interaction (TODO. Is totally deeply embedded in the pyiron library. It's even mangled so atomistics content lives in base. I am not sure how we want to abstract this. We also don't even serialize what code was used (pyiron versions, called executable versions(this maybe we do store?) to produce the output, much less store it nicely in the database.)
+
+For storage, pyiron uses HDF5 and relies on the pyiron/datacontainer library(TODO) which in turn uses the h5io/h5io library.
```

Comment:
At the meeting (2023-01-23) we discussed putting *all* the job input into a single `DataContainer`-powered input field. We had to remind ourselves why [@pmrv thought it was unwise](https://github.com/pyiron/pyiron_atomistics/pull/799#discussion_r994964343) to simply subclass `DataContainer`. We played around briefly with the `pyiron_base.HasStoredTraits` class I made, but it doesn't natively support the nice JSON output that `DataContainer` does (the underlying data container only gets populated *after* a `to_hdf` call in my implementation). @samwaseda also had some concerns about how easy (or not!) it would be to create deeply nested input structures in this way, since `HasStoredTraits` also loses the `create_group` method that he likes to use.